crates/bindings-csharp/Codegen.Tests/TestInit.cs

```diff
@@ -1,14 +1,69 @@
 namespace SpacetimeDB.Codegen.Tests;
 
 using System.Runtime.CompilerServices;
+using System.Text;
+using Microsoft.CodeAnalysis;
 
 // Global Verify setup for all tests we might have.
 static class TestInit
 {
+    // A custom Diagnostic converter that does 2 things that Verify.SourceGenerators doesn't:
+    // 1. Pretty-prints the error with the source code snippet and squiggly underline.
+    // 2. Normalizes the file path to avoid differences between platforms.
+    class DiagConverter : WriteOnlyJsonConverter<Diagnostic>
+    {
+        public override void Write(VerifyJsonWriter writer, Diagnostic diag)
+        {
+            writer.WriteStartObject();
+            // Pretty-print the error with the source code snippet.
+            var loc = diag.Location;
+            if (loc.SourceTree is { } source)
+            {
+                var comment = new StringBuilder().AppendLine();
+                var lineSpan = loc.GetLineSpan();
+                var lines = source.GetText().Lines;
+                const int contextLines = 1;
+                var startLine = Math.Max(lineSpan.StartLinePosition.Line - contextLines, 0);
+                var endLine = Math.Min(
+                    lineSpan.EndLinePosition.Line + contextLines,
+                    lines.Count - 1
+                );
+                for (var lineIdx = startLine; lineIdx <= endLine; lineIdx++)
+                {
+                    var line = lines[lineIdx];
+                    // print the source line
+                    comment.AppendLine(line.ToString());
+                    // print squiggly line highlighting the location
+                    if (line.Span.Intersection(loc.SourceSpan) is { } intersection)
+                    {
+                        comment
+                            .Append(' ', intersection.Start - line.Start)
+                            .Append('^', intersection.Length)
+                            .AppendLine();
+                    }
+                }
+                writer.WriteComment(comment.ToString());
+            }
+            writer.WriteMember(
+                diag,
+                diag.Location.GetMappedLineSpan().ToString().Replace('\\', '/'),
+                nameof(diag.Location)
+            );
+            writer.WriteMember(diag, diag.GetMessage(), "Message");
+            writer.WriteMember(diag, diag.Severity, nameof(diag.Severity));
+            writer.WriteMember(diag, diag.Descriptor, nameof(diag.Descriptor));
+            writer.WriteEndObject();
+        }
+    }
+
     [ModuleInitializer]
     public static void Initialize()
     {
-        VerifierSettings.OmitContentFromException();
+        if (Environment.GetEnvironmentVariable("CI") is null)
+        {
+            // Don't show the full diff in the console when running tests locally as Verify will open the diff tool automatically.
```

Comment:
Will this happen in vscode? 🥺 

I guess I can just use visual studio for editing C# stuff.

Reply:
> Will this happen in vscode? 🥺

Yeah, it just opens whatever default tool you have set up as a diff editor. I think it picks one from Git settings maybe? For me it opens VSCode at least.